### PR TITLE
Migrate GitHub.Network API spec to public repo.

### PR DESCRIPTION
**Pull request by mlacouture**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - Making GitHub.Network RP API spec public so customers can build SDK-like libraries. The RP was released to GA back in April 2024. 

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by mlacouture** [comment]
TypeSpec validation is failing with the following error in a TypeSpec library, can someone help? thanks

image

**Comment by rkmanda** [comment]
> TypeSpec validation is failing with the following error in a TypeSpec library, can someone help? thanks
> 
> image

pl post the question in the TypeSpec teams channel or attend the TypeSpec office hours

**Comment by rkmanda** [comment]
Pl fix these errors or add a suppression indicating why these wont be fixed.. 
https://github.com/Azure/azure-rest-api-specs/pull/30837/checks?check_run_id=30996145039

https://github.com/Azure/autorest/blob/main/docs/generate/suppress-warnings.md

File: specification/github-network/resource-manager/readme.md
Comment (mlacouture) [inline specification/github-network/resource-manager/readme.md]: These versions are blocked by the service if traffic is sent to it, so no customer is using any of them.

File: specification/github-network/GitHub.Network.Management/NetworkSettingsResource.tsp
Comment (mlacouture) [inline specification/github-network/GitHub.Network.Management/NetworkSettingsResource.tsp]: never enabled, removing now.

**Comment by mentat9** [comment]
@mlacouture - Since your APIs are GA and you aren't creating a new API version, I presume your private specs were already reviewed by ARM. Can you provide a link to the PR that was reviewed and signed off by ARM?

**Comment by mlacouture** [comment]
> @mlacouture - Since your APIs are GA and you aren't creating a new API version, I presume your private specs were already reviewed by ARM. Can you provide a link to the PR that was reviewed and signed off by ARM?

@mentat9 this is the PR for the GA specs:
- https://github.com/Azure/azure-rest-api-specs-pr/pull/17246

File: specification/github-network/resource-manager/GitHub.Network/preview/2024-04-01-preview/GitHub.Network.json
Comment (mlacouture) [reply specification/github-network/resource-manager/GitHub.Network/preview/2024-04-01-preview/GitHub.Network.json]: @mentat9 any thoughts on this one?

**Comment by mlacouture** [comment]
@mentat9 do you mind approving again? had to rebase the branch - thank you!

**Comment by mlacouture** [comment]
Removed `ARMSignedOff` label as this requires re-approval after rebasing the branch and no response from approver.

**Comment by mlacouture** [comment]
This PR still needs to be reviewer-approved (not just `ArmSignedOff`), can someone please approve it? thank you!
cc: @mentat9 @rkmanda @msyyc @qiaozha

**Review by mentat9 — APPROVED** [review]

**Pull request merged**
